src/test/java/org/gaul/s3proxy/S3AwsSdkTest.java

```diff
@@ -0,0 +1,172 @@
+
+        Properties properties = new Properties();
+        ContextBuilder builder = ContextBuilder
+                .newBuilder(provider)
+                .credentials(identity, credential)
+                .modules(ImmutableList.<Module>of(new SLF4JLoggingModule()))
+                .overrides(properties);
+        if (!Strings.isNullOrEmpty(endpoint)) {
+            builder.endpoint(endpoint);
+        }
+        context = builder.build(BlobStoreContext.class);
+        BlobStore blobStore = context.getBlobStore();
+        containerName = createRandomContainerName();
+        blobStore.createContainerInLocation(null, containerName);
+
+        S3Proxy.Builder s3ProxyBuilder = S3Proxy.builder()
+                .blobStore(blobStore)
+                .endpoint(s3Endpoint)
+                .forceMultiPartUpload("true".equalsIgnoreCase(
+                        forceMultiPartUpload));
+        if (s3Identity != null || s3Credential != null) {
+            s3ProxyBuilder.awsAuthentication(s3Identity, s3Credential);
+        }
+        if (keyStorePath != null || keyStorePassword != null) {
+            s3ProxyBuilder.keyStore(
+                    Resources.getResource(keyStorePath).toString(),
+                    keyStorePassword);
+        }
+        if (virtualHost != null) {
+            s3ProxyBuilder.virtualHost(virtualHost);
+        }
+        s3Proxy = s3ProxyBuilder.build();
+        s3Proxy.start();
+        while (!s3Proxy.getState().equals(AbstractLifeCycle.STARTED)) {
+            Thread.sleep(1);
+        }
+
+        // reset endpoint to handle zero port
+        s3Endpoint = new URI(s3Endpoint.getScheme(), s3Endpoint.getUserInfo(),
+                s3Endpoint.getHost(), s3Proxy.getPort(), s3Endpoint.getPath(),
+                s3Endpoint.getQuery(), s3Endpoint.getFragment());
+
+        Properties s3Properties = new Properties();
+        s3Properties.setProperty(Constants.PROPERTY_TRUST_ALL_CERTS, "true");
+        s3Context = ContextBuilder
+                .newBuilder("s3")
+                .credentials(s3Identity, s3Credential)
+                .endpoint(s3Endpoint.toString())
+                .overrides(s3Properties)
+                .build(BlobStoreContext.class);
+        s3BlobStore = s3Context.getBlobStore();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        if (s3Proxy != null) {
+            s3Proxy.stop();
+        }
+        if (s3Context != null) {
+            s3Context.close();
+        }
+        if (context != null) {
+            context.getBlobStore().deleteContainer(containerName);
+            context.close();
+        }
+    }
+
+    @Test
+    public void testAwsV4Failure() throws Exception {
+        BasicAWSCredentials awsCreds = new BasicAWSCredentials(s3Identity,
+                s3Credential);
+        AmazonS3 client = new AmazonS3Client(awsCreds);
+        client.setEndpoint(s3Endpoint.toString());
+        try {
+            client.getObject(containerName, "foo");
```

Comment:
You need to call `fail()` call here to ensure that `S3Proxy` throws the expected exception.